### DISCUSSION TITLE
Prisma VSCode extension doesn't auto-complete. Not sure what I'm doing wrong.

### DISCUSSION THREAD

**Discussion by aberkow** [prisma/prisma#5982] [Q&A] [answered]
I'm working on a project with nextJS and prisma and trying to understand what I perceive to be inconsistent behavior from the official VSCode extension. In the generated `prisma` directory, I created a `PrismaClient` and then export it. Inside that file, all the VSCode auto-completion/suggestion works. However, when I import the client into my graphql API, the auto-completion/suggestion stops working. I did this because according to the [documentation on "Instantiating the Client"](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client/#use-a-single-shared-instance-of-prismaclient) you should only create one prisma instance.

If I'm missing something extremely obvious, I apologize. I just can't figure out why the extension isn't working as I expect.

If it matters, I'm using
- macOS 10.15.7 Catalina
- VSCode 1.53.2
- prisma extension v2.18.1

Thanks for any help you can offer!

Examples:

```
/prisma
|_ prisma.js
/pages
|_ /api
  |_ graphql.js
```

```ts
// /prisma/prisma.js

import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

export default prisma
```

<img width="634" alt="Screen Shot 2021-03-04 at 4 10 47 PM" src="https://user-images.githubusercontent.com/4681584/110031919-85ead400-7d05-11eb-9a24-ff8fc74be0d5.png">

As opposed to 
```ts
// /pages/api/graphql.js

// other imports for apollo server...
import prisma from '../../prisma/prisma'

const resolvers = {
  Query: {
     users: (_, args, { prisma } ) => {
       // prisma is added via ApolloServer context
      // however auto-complete/suggest doesn't work here
    }
  }
}
```

<img width="591" alt="Screen Shot 2021-03-04 at 4 11 03 PM" src="https://user-images.githubusercontent.com/4681584/110032276-f3970000-7d05-11eb-9394-a6345c5d86ca.png">

**Comment by ryands17** [comment, marked as answer]
@aberkow 👋 

As you're using JavaScript, you would need to follow [this](https://www.prisma.io/docs/support/help-articles/autocompletion-in-graphql-resolvers-with-js) guide to make autocomplete work in the resolver's `context`.

**Reply by aberkow** [reply]
Thank you! That was exactly what I was missing. Not entirely sure how I missed it, but that's a different problem. Maybe I should also just go learn TypeScript as well.... Anyway, thanks again 🙂

**Reply by KrishnaGupta0405** [reply]
brother could you please again share the same link bcz it now showing -> 
![image](https://github.com/user-attachments/assets/cb5aa86c-653e-4475-97ab-062b22c1eafa)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
